src/test/ui/associated-types/associated-types-issue-20346.stderr

```diff
@@ -6,6 +6,30 @@ LL |     is_iterator_of::<Option<T>, _>(&adapter);
    |
    = note: expected type `T`
               found type `std::option::Option<T>`
+   = help: given a type parameter `T` and a method `foo`:
+           ```
+           trait Trait<T> { fn foo(&self) -> T; }
+           ```
+           the only ways to implement method `foo` are:
+           - constrain `T` with an explicit type:
+           ```
+           impl Trait<String> for X {
+               fn foo(&self) -> String { String::new() }
+           }
+           ```
+           - add a trait bound to `T` and call a method on that trait that returns `Self`:
+           ```
+           impl<T: std::default::Default> Trait<T> for X {
+               fn foo(&self) -> T { <T as std::default::Default>::default() }
+           }
+           ```
+           - change `foo` to return an argument of type `T`:
+           ```
+           impl<T> Trait<T> for X {
+               fn foo(&self, x: T) -> T { x }
```

Comment:
this one confused me since it hinges on changes to the trait def as well as the impl